internal/controller/operatorconfigmap_controller.go

```diff
@@ -493,11 +517,25 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI(storageClients *v1al
 	cephFsDriver := &csiopv1a1.Driver{}
 	cephFsDriver.Name = templates.CephFsDriverName
 	cephFsDriver.Namespace = c.OperatorNamespace
-
 	if err := c.createOrUpdate(cephFsDriver, func() error {
 		if err := c.own(cephFsDriver); err != nil {
 			return fmt.Errorf("failed to own csi cephfs driver: %v", err)
 		}
+		// only update during initial creation
+		if cephFsDriver.UID == "" {
```

Comment:
Your condition should be based on the UID (as it is here) and the existence of custom tolerations (missing check)